DEVELOPING.md

```diff
@@ -1,97 +1,130 @@
-4. Set the correct `embrace-android-sdk` version in both `examples/react-native-test-suite/node_modules/embrace-io/android/build.gradle`
+4. Set the correct `embrace-android-sdk` version in `examples/react-native-test-suite/node_modules/embrace-io/android/build.gradle`
 5. Run the app in the normal way
 
-## Testing iOS changes locally
+## Testing against new Embrace iOS SDK versions
 
-### Local artefact
+### Local artifact
 
-You can test changes local changes to the iOS SDK by updating the React Native Embrace project's `podspec` and `Podfile` to point to the local copy.
+You can test changes local changes to the iOS SDK by updating the example app's `podspec` and `Podfile` to point to the local copy.
 
 1. In `examples/react-native-test-suite/node_modules/embrace-io/RNEmbrace.podspec`, change the dependency on the iOS SDK to `s.dependency 'EmbraceIO-LOCAL'`
 2. In `examples/react-native-test-suite/ios/Podfile`, add the following line `pod 'EmbraceIO-LOCAL', :path => 'path/to/ios_sdk'`
 3. In `examples/react-native-test-suite/ios`, run the `pod update` command
 
-### Beta artefact
+### Beta artifact
 
 1. Ask the iOS team to publish a beta of the iOS SDK to the `EmbraceIO-DEV` pod
 2. In `examples/react-native-test-suite/node_modules/embrace-io/RNEmbrace.podspec`, change the dependency on the iOS SDK to `s.dependency 'EmbraceIO-DEV'`
 3. In `examples/react-native-test-suite/ios/Podfile`, add the following line `pod 'EmbraceIO-DEV'`
 4. In `examples/react-native-test-suite/ios`, run the `pod update` command
 
-# Releasing the SDK
+## Automated integration testing
+
+Automated integration testing is being actively developed to replace some of the manual testing outlined above, see [here](./integration-tests/README.md) for more details.
 
-0. Bump the Android (SDK + Swazzler)/iOS dependencies to the latest available stable versions
-   0.1. Update the iOS SDK version in `./RNEmbrace.podspec`
-   0.2. Update the Android SDK version in `./android/gradle.properties`
-1. Bump the SDK version according to semver
-2. Run the example app on Android + iOS (in release mode) and confirm that a session is captured & appears in the dashboard with useful info
-3. Create a PR with all these changes and merge to `master`
-4. Release to npm with `yarn publish --tag <your-version-here>`
+## Branching strategy
 
-# Releasing a beta SDK version
+Generally all proposed changes should target the `master` branch and new releases are cut from there after QA. One exception
+is urgent patch fixes, in those cases a branch should be made from the latest released tag to isolate the fix from any
+unreleased changes on `master` and a patch release will be cut from that new branch.
 
-To release a beta version of the SDK you should add a suffix to the version string. For example, `1.3.0-beta01` would be an appropriate name.
+## Releasing
 
-Otherwise, the release process will be exactly the same as for a regular release.
```

Comment:
@TheMutsi this is the section I'll need the most help with I'm not sure the existing steps were up-to-date, for example I don't see the "publish-modules" script being run

some additional questions:
* is updating the generated docs part of the release process?
* is there a changelog that gets updated as part of the release process?
* are we able to leverage github releases? I see that https://github.com/embrace-io/embrace-react-native-sdk/releases is currently blank

Reply:
Yes the step 5 should be replaced by run
`yarn publish-modules`

The generated doc is something you can ask internally. When I asked it 2 years ago they told me let it for now. But I dont know if is still the same situation. Its pretty useless to me

The changelog should be added in the embrace-docs 

Yes you can, if you check the previous repo, the private one. You are going to find the CI CD for this. We were waiting to define a good PR / merge system before include it here

Reply:
ok great I'll take a look, I'm tempted to remove the docs related stuff then if it's not being used

Reply:
CI/CD integration should be tackled as a separate ticket: https://www.notion.so/embraceio/Integrate-releases-and-changelog-updates-with-CI-CD-ab93efad962d4e6aba692dfb5d9dba81?pvs=4

Reply:
Yarn publish is a reserved method to deploy a monorepo. You shouldn't use that name for modules

Reply:
changed back